src/Redirection/Dynamic/RedirectionManager.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace AppBundle\Redirection\Dynamic;
+
+use AppBundle\Entity\Redirection;
+use Doctrine\ORM\EntityManagerInterface;
+use JMS\Serializer\Serializer;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+
+class RedirectionManager
+{
+    private $cache;
+    private $entityManager;
+    private $serializer;
+
+    public function __construct(AdapterInterface $cache, EntityManagerInterface $entityManager, Serializer $serializer)
+    {
+        $this->cache = $cache;
+        $this->entityManager = $entityManager;
+        $this->serializer = $serializer;
+    }
+
+    public function refreshRedirectionCache(Redirection $redirection): void
+    {
+        $item = $this->cache
+            ->getItem('redirection-'.md5($redirection->getFrom()))
```

Comment:
The prefixed is not required if you have a dedicated cache pool. And I don't think the `md5` is needed. The key of the item could be any string I guess.